eng/Analyzers.props

```diff
@@ -3,9 +3,10 @@
     <CodeAnalysisRuleset>$(MSBuildThisFileDirectory)CodeAnalysis.ruleset</CodeAnalysisRuleset>
   </PropertyGroup>
   <ItemGroup Condition="'$(EnableAnalyzers)' == 'true'">
-    <PackageReference Include="Microsoft.DotNet.CodeAnalysis" Version="$(MicrosoftDotNetCodeAnalysisVersion)" />
-    <PackageReference Include="Microsoft.CodeAnalysis.Common" Version="3.4.0-beta2-final" />
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.6" />
-    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.66" />
+    <PackageReference Include="Microsoft.DotNet.CodeAnalysis" Version="$(MicrosoftDotNetCodeAnalysisVersion)" PrivateAssets="all" IsImplicitlyDefined="true" />
+    <!-- Microsoft.CodeAnalysis.Common brings in external dependencies that CoreLib can't deal with. -->
```

Comment:
How does "can't deal with" manifest?

Reply:
Thousands of the following errors:

```
C:\git\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\MemoryFailPoint.cs(368,128): error CS0518: Predefined type 'System.Boolean' is not defined or imported [C:\git\runtime\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
C:\git\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\MemoryFailPoint.cs(368,156): error CS0518: Predefined type 'System.UInt64' is not defined or imported [C:\git\runtime\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
C:\git\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\MemoryFailPoint.cs(368,177): error CS0518: Predefined type 'System.UInt64' is not defined or imported [C:\git\runtime\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
C:\git\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\MemoryFailPoint.cs(368,206): error CS0518: Predefined type 'System.Int64' is not defined or imported [C:\git\runtime\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
C:\git\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\MemoryFailPoint.cs(368,238): error CS0518: Predefined type 'System.UInt64' is not defined or imported [C:\git\runtime\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
C:\git\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\MemoryFailPoint.cs(368,13): error CS0518: Predefined type 'System.Void' is not defined or imported [C:\git\runtime\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
src\System\RuntimeArgumentHandle.cs(15,23): error CS0518: Predefined type 'System.ValueType' is not defined or imported [C:\git\runtime\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
```

AFAIK CoreLib mustn't have any references. That packages brings some in. I'm right now experimenting with adding a netstandard2.1 configuration to `Microsoft.CodeAnalysis.csproj`.

Reply:
The analyzer dependencies are not supposed to end up as `/r:` references for the compiler. This means that either this package is not authored correctly; or it is not used correctly.

Reply:
> or it is not used correctly.

It shouldn't be referenced directly. The entry was added because libraries inject the analyzers into the csc task in a special way (no project restore). This should now be resolved.